### PR TITLE
proto: add cleanup, rollback_get, commit_get proto

### PR DESCRIPTION
**Pull request by morefreeze**
`codec.go` is migrated from rust server to golang
@disksing PTAL

### CODE REVIEW

File: kvrpc/proto/kvrpcpb.proto [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    CmdScan = 2;
+    CmdPrewrite = 3;
+    CmdCommit = 4;
+    CmdCleanUp = 5;
+    CmdRollbackThenGet = 6;
+    CmdCommitThenGet = 7;
+}
+
+message KvPair {
+    required bytes key = 1;
```
Comment (ngaut) [inline kvrpc/proto/kvrpcpb.proto]: Do not use required annotate.

File: kvrpc/codec.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	// MsgMagic a fix identifier of message
+	MsgMagic uint16 = 0xdaf4
+	// MsgVersionV1 message version 1
+	MsgVersionV1 uint16 = 1
+	_msgHeader          = RPCMsgHeader{}
+)
+
+// MsgHeaderSize sizeof(RPCMsgHeader)
+// Check this value if migrate other OS, like Windows.
+const MsgHeaderSize int = int(unsafe.Sizeof(_msgHeader))
```
Comment (disksing) [inline kvrpc/codec.go]: Maybe `const MsgHeaderSize int = int(unsafe.Sizeof(RPCMsgHeader{}))` and discard `_msgHeader`.

**Comment by siddontang** [comment]
maybe we should use `https://github.com/gogo/protobuf` instead 

@ngaut

File: kvrpc/proto/kvrpcpb.proto (hunk 1 above)
Comment (morefreeze) [inline kvrpc/proto/kvrpcpb.proto]: @ngaut s/required/optional/g ?

**Comment by ngaut** [comment]
Later @siddontang 
Currently rust-protobuf can't recognize gogo annotate

**Comment by ngaut** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**